### ISSUE TITLE
reapchild failed with errno=10 No child processes

### ISSUE THREAD

**Issue by dkg** [libreswan/libreswan#72]
When started under systemd, pluto produces this output in the journal:

```
Aug 07 23:37:23 machine pluto[7748]: reapchild failed with errno=10 No child processes
```

presumably something is calling reapchild when it shouldn't be called

**Comment by letoams** [comment]
Yes, the old adns helper thread...

**Comment by dkg** [comment]
On Sun 2016-08-07 23:46:17 -0400, Paul Wouters (libreswan) wrote:
>> On Aug 7, 2016, at 11:41 PM, dkg <notifications@github.com> wrote:
>> 
>> When started under systemd, pluto produces this output in the journal:
>> 
>> Aug 07 23:37:23 machine pluto[7748]: reapchild failed with errno=10 No child processes
>> presumably something is calling reapchild when it shouldn't be called
>
> Yes, the old adns helper thread...
> Sent from my iPhone

I don't know what this means or how to resolve the problem.  can you be
more specific?

     --dkg

**Comment by kirill-krokodil** [comment]
hello, @letoams , seems like some people encountering the same issue.
for example, i came here from this tutorial: https://www.moucha.cc/charon/l2tpipsec-vpn-client-on-fedora/
where another 3 users (+me) couldn't resolve the problem.

may be you can say, what is wrong and how to fix it?
thank you!

**Comment by letoams** [comment]
You can ignore this error. It is the result of us no longer running adns_helper processes, which the main process is still trying to reap. It is not affecting connections whatsoever, so if you have problems with connections, this is not related to your problem.

**Comment by kirill-krokodil** [comment]
@letoams 
aha, thank you for answering. but after getting this error terminal does not give back control to the user, just waiting for somewhat, so i should stop it manually (ctrl+c).
may be, that is why users so frustrated - process breaks at that place - just after getting this error message.

**Comment by letoams** [comment]
I the link you showed starts ipsec in the foreground on purpose using $ ipsec pluto --stderrlog --config /etc/ipsec.conf --nofork so yes then it will never daemonize and run in the background. It will also skip all the PreStart= tasks we do. The proper way to start the libreswan ipsec service is to the use the systemctl or service command to start the service. If you run "ipsec start" it will autodetect your init system and call the right command (eg systemctl start ipsec.service on systemd)

**Comment by kirill-krokodil** [comment]
@letoams i've got it, thank you!

**Comment by letoams** [comment]
fixed in 3.19
